layouts/PostLayout.tsx

```diff
@@ -20,9 +21,41 @@ interface LayoutProps {
   children: ReactNode;
 }
 
+const u = (url = '', params: Record<string, string> = {}) => {
+  let result = new URL(url, 'https://tailscale.dev');
+  Object.entries(params).forEach((kv) => {
+    let [k, v] = kv;
+    result.searchParams.set(k, v);
+  });
+  return result.toString();
+};
+
+const ShareLinkButton = ({
```

Comment:
Can probably be lifted to separate component file, it's only a function in this file because of the rule of 3's. I use this button five times.

Reply:
I vote for removing the duplication, which will make it easier to iterate on the look and feel of the elements. Can either have an object you iterate over or extract it into a component.